_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,91 @@
+
+This behaviour is bad because creative people have the habit to question rules if they are in their way. If the result is that the rule does not create value (but supresses value-creation) in the case the engineer, striving to achieve a goal, will try to navigate around it. In such a case: __question the rule__, not the violator. In fact violating/questioning rules is a __healthy habit__.
+
+Feedback should be stated like this: _At some point we decided to approach things like this like that because we were convinced that's better because of reason X and Y. Don't you agree? Shall we re-evaluate?_
+
+Usually the topic is done now. The reviewee will probably accept the suggestion because the reviewer acted in a consulting role and took the reviewee serious, asked for his opinion.
+
+The reviewee will supposely not feel that the rule harms him so badly any more and follow it. Or a healthy discussion arises that in the end might change something for the better for the whole team.
+
+As a general rule of thumb: __do not define too many rules and processes__. And in a review: start with understanding the requirements, think about potential solutions also from a high-level/architectural/design-perspective. __How would you have solved the problem?__ Then look at the solution. __Find real bugs__. Only beginners focus on cosmetics (and usually miss real flaws).
+
+## Cosmetics Are Cosmetics
+
+Cosmetics are called cosmetics because they are __not substantial__. Bad formatting, incomplete documentation, bad naming or other trivial things should be handled as what they are: cosmetics. Don't ignore them, __good make-up can make a difference__. But as a reviewer, double-check to make sure your attitude fits the importance of the advice. Nothing is worse than bad vibes because of things that do not have a real impact.
+
+## Don't Override - Contribute
+
+It is generally not a bad idea to fix small issues during the code review. You can still point the reviewee at the changes but it does not make sense to invest the time twice. If you can fix a spelling error on the fly - why not do it?
+
+But don't change the design completely. Do not override the __substance__ of what your reviewee wanted to do. Help him, don't control him.
+
+
+## Summing up
+
+It is all about your __mindset__ and __team dynamics__. People need to resist the urge to use the power they have in a review situation. Teams need to come to a state in which __openness__ and __transparency__ are worshipped as __virtues__. Into a state in which being asked for a review is a compliment that everyone is happy to take. A state in which everyone treats the others weaknesses as valuable as their strengths and actually wants to get better together. You do not need a book of behaviour. You do not need strict rules for a review. What you need is  respect for others and some emotional intelligence to anticipate how they will __feel__ reading or listening to your review comments.
```

Comment:
Good stuff.

If I may add my own content suggestion. I've found that as a reviewer, if I give compliments to someone's implementation then it can lessen the unwanted negative or criticizing aspects of a review. Something like: "nice solution" or "neat" when merited, can reinforce a positive mindset in reviews and communicate to the author that the reviewer is truly interested in solving the problem, rather than simply solving the problem **their way**. 

Additionally, this may have been deliberate, but the blog mostly focuses on the behavior of the reviewer, rather than the author in a review. I think there's definitely some points to be made to guide authors when responding to review in order to create a constructive environment; such how to respond to feedback and practices like using small pull requests. If you agree, perhaps that be added here or maybe even a second post?